app/src/debug/java/timur/gilfanov/messenger/data/source/remote/RemoteDataSourceFake.kt

```diff
@@ -39,23 +40,41 @@ class RemoteDataSourceFake @Inject constructor() :
         private const val SENDING_PROGRESS_COMPLETE = 100
     }
 
-    private val serverChatsFlow = MutableStateFlow<Map<ChatId, Chat>>(emptyMap())
-    private val connectionStateFlow = MutableStateFlow(true)
+    /**
+     * Unified server state to prevent race conditions between chats and operations.
+     * Tracks lastSyncTimestamp to ensure deterministic behavior while preventing sync race conditions.
+     */
+    private data class ServerState(
+        val chats: Map<ChatId, Chat> = emptyMap(),
+        val operationTimestamps: Map<String, Instant> = emptyMap(),
+        val currentTimestamp: Instant = Instant.fromEpochMilliseconds(0),
+        val lastSyncTimestamp: Instant = Instant.fromEpochMilliseconds(0),
+    ) {
+        fun nextTimestamp(): Instant = currentTimestamp.plus(1.seconds)
```

Comment:
Not used outside ServerState, can be private.
```suggestion
        private fun nextTimestamp(): Instant = currentTimestamp.plus(1.seconds)
```